### PR TITLE
Sustain service broker

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- implement controllers which are responsible for (cluster)servicebroker management (creation and deletion)
- add component tests for provisioning/deprovisioning

**Related issue(s)**
Related: #54

### CODE REVIEW

File: internal/controller/cluster_addons_controller_test.go [hunk 1]
```diff
@@ -73,6 +75,13 @@ func TestReconcileClusterAddonsConfiguration_AddAddonsProcess(t *testing.T) {
 
 }
 
+// newEmptyInstanceChecker returns instanceChecker which behaves like there is no any service instances in the cluster.
+func newEmptyInstanceChecker(t *testing.T) instanceChecker {
```
Comment (jasiu001) [inline internal/controller/cluster_addons_controller_test.go]: it is not used anywhere

File: internal/controller/instance/facade.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			if _, exists := appBrokerClassNames[inst.Spec.ServiceClassRef.Name]; exists {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// AnyServiceInstanceExistsForClusterServiceBroker checks whether there is at least one service instance created with helm broker cluster service class.
+func (f *Facade) AnyServiceInstanceExistsForClusterServiceBroker(namespace string) (bool, error) {
```
Comment (jasiu001) [inline internal/controller/instance/facade.go]: `namespace` is not neccessary

File: internal/controller/broker_controller.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &v1alpha1.AddonsConfiguration{}}, eventHandler, createDeletePredicate)
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &v1beta1.ServiceBroker{}}, eventHandler, predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool { return e.Meta.GetName() == broker.NamespacedBrokerName },
```
Comment (jasiu001) [inline internal/controller/broker_controller.go]: why here is pernament name `helm-broker` and cluster is set by configuration?

Comment (piotrmiskiewicz) [reply internal/controller/broker_controller.go]: the name of the namesaces ServiceBroker is set as a constant in the code. The ClusterServiceBroker is configurable

**Pull request merged**